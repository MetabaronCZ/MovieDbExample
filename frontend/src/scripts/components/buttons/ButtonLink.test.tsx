import { describe, expect, it } from '@jest/globals';
import { getByRole } from '@testing-library/dom';

import { ButtonLink } from 'components/buttons/ButtonLink';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/button/ButtonLink', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <ButtonLink text="CONTENT" to="/test" />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const link = getByRole(container, 'link');
    expect(link).toBeInTheDocument();
    expect(link).toHaveTextContent('CONTENT');
    expect(link).toHaveAttribute('href', '/test');
  });

  it('should render ico before', async () => {
    const { container } = await waitForComponent(
      <ButtonLink text="CONTENT" icoBefore="cross" to="/test" />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const link = getByRole(container, 'link');
    expect(link).toBeInTheDocument();

    const ico = link.querySelector('svg.fa-xmark');
    expect(ico).toBeInTheDocument();
  });

  it('should render ico after', async () => {
    const { container } = await waitForComponent(
      <ButtonLink text="CONTENT" icoAfter="cross" to="/test" />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const link = getByRole(container, 'link');
    expect(link).toBeInTheDocument();

    const ico = link.querySelector('svg.fa-xmark');
    expect(ico).toBeInTheDocument();
  });
});
