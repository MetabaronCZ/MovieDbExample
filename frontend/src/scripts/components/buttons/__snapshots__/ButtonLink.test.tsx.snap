// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/button/ButtonLink should render ico after 1`] = `
.c2 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c0:hover {
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  text-decoration: none;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

<div>
  <a
    class="c0 c1"
    data-discover="true"
    href="/test"
  >
    <span
      class="c2"
    >
      CONTENT
    </span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-xmark fa-sm "
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`components/button/ButtonLink should render ico before 1`] = `
.c2 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c0:hover {
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  border: 2px solid #002d56;
  background: #fff;
  text-decoration: none;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

<div>
  <a
    class="c0 c1"
    data-discover="true"
    href="/test"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-xmark fa-sm "
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
      />
    </svg>
    <span
      class="c2"
    >
      CONTENT
    </span>
  </a>
</div>
`;

exports[`components/button/ButtonLink should render properly 1`] = `
.c2 {
  flex: 1;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c0 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c0:hover {
  text-decoration: none;
}

.c0:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  border: 2px solid #002d56;
  background: #fff;
  text-decoration: none;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

<div>
  <a
    class="c0 c1"
    data-discover="true"
    href="/test"
  >
    <span
      class="c2"
    >
      CONTENT
    </span>
  </a>
</div>
`;
